### PR TITLE
Template: docker_build_ci.yml

### PR DESCRIPTION
**Pull request by pascalandy**
Following https://github.com/actions/starter-workflows/issues/59.

I had a hard time to understand the syntax of a real life CI scenario looking at the [actual docker template](https://github.com/actions/starter-workflows/blob/master/ci/docker-image.yml).

After many hours reading docs, looking at other repo using actions I found a decent way to bring the CI to life (and on part with Travis CI).

With this template, people will be able to quickly modify the yaml and make it their own.

### Variable

We can see that I fetch VARs from the Dockerfile itself. If you guys prefer, I could hardcode them all if it's easier for everyone to consume the template.

Cheers!

### CODE REVIEW

**Comment by 1138-4EB** [comment]
Ref #96

File: ci/docker_build_ci.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+name: docker_build_ci
+on:
+  push:
+  # everyday at 3 am
+  schedule:
+    - cron:  '0 3 * * *'
```
Comment (andymckay) [inline ci/docker_build_ci.yml]: We don't put `schedule` in our starter workflow, preferring either `push` or `release`.

**Comment by pascalandy** [comment]
@andymckay sure, I understand. It's all about giving a decent template to build Dockerfiles.

What are your specs for a good template? IMHO, at the moment, the template is not useful at all.

**Comment by 1138-4EB** [comment]
> What are your specs for a good template?

Maybe just add the following to https://github.com/actions/starter-workflows/blob/master/ci/docker-image.yml?

```
  run: |
    echo "$DOCKER_PASS" | docker login -u "$DOCKER_USER" --password-stdin
    docker push my-image-name:$(date +%s)
    docker logout
  env:
    DOCKER_USER: ${{ secrets.DOCKER_USER }}
    DOCKER_PASS: ${{ secrets.DOCKER_PASS }}
```

Note #96.

**Comment by pascalandy** [comment]
This is where we disagree. Personally, I want to see a 'advance' template as well so I can see what are my options. Many of us a power users over here right? 

> but as a starter workflow is likely to cause confusion and has an awful lot of information that isn't relevant to many users. It looks like a great blog post or how-to article somewhere else though.

**Comment by pascalandy** [comment]
For everyone follow this, you might be interested by our [template over here](https://github.com/firepress-org/ghostfire/tree/master/.github/workflows). It's the core one we use for all our github actions project.

**Pull request closed without merging**